File: php/Scheme/Matrice/GASM.php
<?php

namespace scheme\matrice;

use Scheme\Interval;
use Scheme\Matrice\AGenAllSchemeMatrices;
use Scheme\Formula\Interval\ListInterval;

class GASM extends AGenAllSchemeMatrices {
                        
  private $schemes;

  public function __construct($nbNum, $schemeSize) {
    parent::__construct($nbNum, $schemeSize);
    $this->schemes = new ListInterval();
    $this->schemes->add(new Interval($this->firstScheme));
  }
                
  public function getSchemes() {                        
    $this->addScheme();                    
    return $this->schemes;
  }             
        
  protected function addScheme() {
    $scheme = $this->firstScheme;
                        
    while ($scheme[0] <= $this->nbNum - $this->schemeSize) {       
      $scheme = $this->genScheme($scheme);
      $this->schemes->add(new Interval($scheme));                              
    }
  }                     
}

File: java/src/matrice/GASM.java
package matrice;

import scheme.Interval;
import formula.interval.ListInterval;

public class GASM extends AGenAllSchemeMatrices {
			
		private ListInterval schemes;
	
		protected void addScheme() {
			byte[] scheme = firstScheme;
			
			while (scheme[0] <= nbNum - schemeSize) {	
				scheme = genScheme(scheme);
				schemes.add(new Interval(scheme));				
			}
		}
			
		public long getSize(int nbNum, int schemeSize) {		
			double prob = 1.0;
			
			for (int i = 1; i <= schemeSize; i++) {
				prob *= ((double)i/ (nbNum - (schemeSize - i )));
			}
			
			return (long) (1/prob);
		}
		
		public ListInterval getSchemes() {
			
			addScheme();			
			return this.schemes;
		}	
				
		public GASM(int nbNum, int schemeSize) {
			super(nbNum, schemeSize);
			schemes = new ListInterval();
			schemes.add(new Interval(firstScheme));
		}
	}

